pkg/provider/kind/kind.go

```diff
@@ -0,0 +1,194 @@
+	DeploymentVars map[string]string
+	// DeployResource to construct DeploymentVars and DeploymentFiles
+	DeploymentResource *provider.DeploymentResource
+	// Content bytes after parsing the template variables, grouped by filename.
+	kindResources []Resource
+	// K8s resource.runtime objects after parsing the template variables, grouped by filename.
+	k8sResources []k8sProvider.Resource
+
+	ctx context.Context
+}
+
+func New(dr *provider.DeploymentResource) *KIND {
+	return &KIND{
+		DeploymentResource: dr,
+		kindProvider: cluster.NewProvider(
+			cluster.ProviderWithLogger(cmd.NewLogger()),
+		),
+		ctx: context.Background(),
+	}
+}
+
+// SetupDeploymentResources Sets up DeploymentVars and DeploymentFiles
+func (c *KIND) SetupDeploymentResources(*kingpin.ParseContext) error {
+	c.DeploymentResource.DefaultDeploymentVars["NGINX_SERVICE_TYPE"] = "NodePort"
+	c.DeploymentResource.DefaultDeploymentVars["LOADGEN_SCALE_UP_REPLICAS"] = "2"
+	c.DeploymentFiles = c.DeploymentResource.DeploymentFiles
+	c.DeploymentVars = provider.MergeDeploymentVars(
+		c.DeploymentResource.DefaultDeploymentVars,
+		c.DeploymentResource.FlagDeploymentVars,
+	)
+	return nil
+}
+
+// KINDDeploymentsParse parses the environment/kind deployment files and saves the result as bytes grouped by the filename.
+// Any variables passed to the cli will be replaced in the resources files following the golang text template format.
+func (c *KIND) KINDDeploymentsParse(*kingpin.ParseContext) error {
+	deploymentResource, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		log.Fatalf("Couldn't parse deployment files: %v", err)
+	}
+	c.kindResources = deploymentResource
+	return nil
+}
+
+func (c *KIND) K8SDeploymentsParse(*kingpin.ParseContext) error {
+	deploymentResource, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		log.Fatalf("Couldn't parse deployment files: %v", err)
+	}
+	for _, deployment := range deploymentResource {
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		k8sObjects := make([]runtime.Object, 0)
+
+		for _, text := range strings.Split(string(deployment.Content), provider.Separator) {
+			text = strings.TrimSpace(text)
+			if len(text) == 0 {
+				continue
+			}
+
+			resource, _, err := decode([]byte(text), nil, nil)
+			if err != nil {
+				return errors.Wrapf(err, "decoding the resource file:%v, section:%v...", deployment.FileName, text[:100])
+			}
+			if resource == nil {
+				continue
+			}
+			k8sObjects = append(k8sObjects, resource)
+		}
+		if len(k8sObjects) > 0 {
+			c.k8sResources = append(c.k8sResources, k8sProvider.Resource{FileName: deployment.FileName, Objects: k8sObjects})
+		}
+	}
+	return nil
+}
+
+// ClusterCreate create a new cluster or applies changes to an existing cluster.
+func (c *KIND) ClusterCreate(*kingpin.ParseContext) error {
+	clusterName, ok := c.DeploymentVars["CLUSTER_NAME"]
```

Comment:
I think we could have `checkDeploymentVarsAndFiles` function and use it for these checks. :)

https://github.com/prometheus/test-infra/blob/21811bc9efd1792478c0b6b81f9ec12a30f9537c/pkg/provider/gke/gke.go#L197-L209

Reply:
Okay, Will add this function

Reply:
Done

Reply:
We should be able to remove this check for variable existence because we're already calling `checkDeploymentVarsAndFiles` when parsing the file.